internal/service/secret.go

```diff
@@ -96,12 +100,58 @@ func (s *secretService) CreateAPIKey(ctx context.Context, req *dto.CreateAPIKeyR
 	// Hash the entire API key for storage
 	hashedKey := s.encryptionService.Hash(apiKey)
 
-	// Set default permissions if none provided
-	permissions := req.Permissions
-	if len(permissions) == 0 {
-		permissions = []string{"read", "write"}
+	// Determine which user to get roles from
+	userID := req.ServiceAccountID
+	if userID == "" {
+		// No service_account_id provided - use authenticated user from context
+		userID = types.GetUserID(ctx)
 	}
 
```

Comment:
if still user id is empty ie not in request and not from context - throw error directly